### PR TITLE
Le Chiffre rev 2 and HE additions

### PR DESCRIPTION
**Pull request by tominabox1**
## Description

Updated  Le Chiffre keyboard folder to add the rev2 version from P3D and the Hotswap Edition from Checkerboards.xyz

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/tominabox1/le_chiffre/he/rules.mk [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (drashna) [inline keyboards/tominabox1/le_chiffre/he/rules.mk:11]:
```suggestion
```

File: keyboards/tominabox1/le_chiffre/he/rules.mk [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+BOOTLOADER = atmel-dfu
+
+BOOTMAGIC_ENABLE = yes     # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes           # USB Nkey Rollover
```
Comment (drashna) [inline keyboards/tominabox1/le_chiffre/he/rules.mk:18]:
```suggestion
NKRO_ENABLE = yes           # Enable N-Key Rollover
```

File: keyboards/tominabox1/le_chiffre/info.json [hunk 3]
```diff
@@ -2,6 +2,8 @@
   "keyboard_name": "le_chiffre",
   "url": "",
   "maintainer": "tominabox1",
+  "width": 11,
+  "height": 4,
```
Comment (drashna) [inline keyboards/tominabox1/le_chiffre/info.json:6]:
```suggestion
```

File: keyboards/tominabox1/le_chiffre/keymaps/default/keymap.c [hunk 4]
```diff
@@ -55,7 +55,7 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
   ),
 
   [_NAV] = LAYOUT(
-      QK_BOOT,  _______,  AG_NORM,  AG_SWAP,  DEBUG, KC_TRNS,   KC_GRV,  KC_PGDN,    KC_UP,  KC_PGUP,  KC_SCLN,
+      RESET,  _______,  AG_NORM,  AG_SWAP,  DEBUG, KC_TRNS,   KC_GRV,  KC_PGDN,    KC_UP,  KC_PGUP,  KC_SCLN,
```
Comment (drashna) [inline keyboards/tominabox1/le_chiffre/keymaps/default/keymap.c:58]: Please revert. `RESET` is being deprecated, in favor of a keycode that doesn't collide with controller specific code.

File: keyboards/tominabox1/le_chiffre/keymaps/default/keymap.c [hunk 5]
```diff
@@ -185,15 +185,13 @@ void render_mod_status(uint8_t modifiers) {
     oled_write_ln_P(PSTR("GUI"), (modifiers & MOD_MASK_GUI));
 }
 
-bool oled_task_user(void) {
+void oled_task_user(void) {
```
Comment (drashna) [inline keyboards/tominabox1/le_chiffre/keymaps/default/keymap.c:188]: Please revert. 
```suggestion
bool oled_task_user(void) {
```

File: keyboards/tominabox1/le_chiffre/keymaps/default/keymap.c [hunk 6]
```diff
@@ -185,15 +185,13 @@ void render_mod_status(uint8_t modifiers) {
...
     render_lechiffre_logo();
     oled_set_cursor(0,3);
     render_layer_status();	// Renders the current keyboard state (layer, lock, caps, scroll, etc)
 	render_mod_status(get_mods()|get_oneshot_mods());
 	render_keylock_status(host_keyboard_led_state());
 	render_keylogger_status();
-
```
Comment (drashna) [inline keyboards/tominabox1/le_chiffre/keymaps/default/keymap.c:195]: Please revert

File: keyboards/tominabox1/le_chiffre/keymaps/via/keymap.c [hunk 7]
```diff
@@ -144,14 +144,13 @@ void render_mod_status(uint8_t modifiers) {
     oled_write_ln_P(PSTR("GUI"), (modifiers & MOD_MASK_GUI));
 }
 
-bool oled_task_user(void) {
+void oled_task_user(void) {
```
Comment (drashna) [inline keyboards/tominabox1/le_chiffre/keymaps/via/keymap.c:147]: please revert.

File: keyboards/tominabox1/le_chiffre/rev1/rules.mk [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = atmel-dfu
```
Comment (drashna) [inline keyboards/tominabox1/le_chiffre/rev1/rules.mk:12]:
```suggestion
BOOTLOADER = atmel-dfu
```

File: keyboards/tominabox1/le_chiffre/rev1/rules.mk [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+
+BOOTMAGIC_ENABLE = yes     # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes           # USB Nkey Rollover
```
Comment (drashna) [inline keyboards/tominabox1/le_chiffre/rev1/rules.mk:18]:
```suggestion
NKRO_ENABLE = yes           # Enable N-Key Rollover
```

File: keyboards/tominabox1/le_chiffre/rev2/rules.mk [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (drashna) [inline keyboards/tominabox1/le_chiffre/rev2/rules.mk:11]:
```suggestion
```

File: keyboards/tominabox1/le_chiffre/rev2/rules.mk [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+BOOTLOADER = atmel-dfu
+
+BOOTMAGIC_ENABLE = yes     # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+NKRO_ENABLE = yes           # USB Nkey Rollover
```
Comment (drashna) [inline keyboards/tominabox1/le_chiffre/rev2/rules.mk:18]:
```suggestion
NKRO_ENABLE = yes           # Enable N-Key Rollover
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/tominabox1/le_chiffre/keymaps/default/keymap.c (hunk 6 above)
Comment (tominabox1) [reply keyboards/tominabox1/le_chiffre/keymaps/default/keymap.c:195]: Ah I see I need a return false... updated!

File: keyboards/tominabox1/le_chiffre/keymaps/via/keymap.c [hunk 12]
```diff
@@ -144,13 +144,14 @@ void render_mod_status(uint8_t modifiers) {
     oled_write_ln_P(PSTR("GUI"), (modifiers & MOD_MASK_GUI));
 }
 
-bool oled_task_user(void) {
+void oled_task_keymap(void) {
```
Comment (drashna) [inline keyboards/tominabox1/le_chiffre/keymaps/via/keymap.c:147]: Pretty sure this is supposed to be:
```suggestion
bool oled_task_user(void) {
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by tominabox1** [comment]
Any updates?

**Review by drashna — APPROVED** [review]

File: keyboards/tominabox1/le_chiffre/readme.md [hunk 13]
```diff
@@ -4,12 +4,18 @@
...
-* Hardware Supported: Le Chiffre (source not available at this time)
+* Hardware Supported: [Le Chiffre](https://github.com/tominabox1/Le-Chiffre-Keyboard)
 * Hardware Availability: Bring your own case.
 
 Make example for this keyboard (after setting up your build environment):
 
-    make tominabox1/le_chiffre:default
+Rev 1:    make tominabox1/le_chiffre:default
+Rev 2:    make tominabox1/le_chiffre/rev2:default
+HE:       make tominabox1/le_chiffre/he:default
```
Comment (fauxpark) [inline keyboards/tominabox1/le_chiffre/readme.md:19]:
```suggestion
    make tominabox1/le_chiffre/rev1:default
    make tominabox1/le_chiffre/rev2:default
    make tominabox1/le_chiffre/he:default
```

File: keyboards/tominabox1/le_chiffre/he/rules.mk [hunk 14]
```diff
@@ -0,0 +1,18 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+NKRO_ENABLE = yes           # USB Nkey Rollover
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+TAP_DANCE_ENABLE = no
+ENCODER_ENABLE = yes
+OLED_ENABLE = no
+RGB_MATRIX_ENABLE = yes
+RGB_MATRIX_DRIVER = WS2812
+LTO_ENABLE = yes
+
```
Comment (fauxpark) [inline keyboards/tominabox1/le_chiffre/he/rules.mk:18]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = yes           # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
ENCODER_ENABLE = yes
RGB_MATRIX_ENABLE = yes
RGB_MATRIX_DRIVER = WS2812
LTO_ENABLE = yes
```

File: keyboards/tominabox1/le_chiffre/rev1/rules.mk [hunk 15]
```diff
@@ -0,0 +1,18 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+NKRO_ENABLE = yes           # USB Nkey Rollover
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+TAP_DANCE_ENABLE = no
+ENCODER_ENABLE = yes
+OLED_ENABLE = no
+RGB_MATRIX_ENABLE = yes
+RGB_MATRIX_DRIVER = WS2812
+LTO_ENABLE = yes
+
```
Comment (fauxpark) [inline keyboards/tominabox1/le_chiffre/rev1/rules.mk:18]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = yes           # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
ENCODER_ENABLE = yes
RGB_MATRIX_ENABLE = yes
RGB_MATRIX_DRIVER = WS2812
LTO_ENABLE = yes
```

File: keyboards/tominabox1/le_chiffre/rev2/rules.mk [hunk 16]
```diff
@@ -0,0 +1,18 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+NKRO_ENABLE = yes           # USB Nkey Rollover
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+TAP_DANCE_ENABLE = no
+ENCODER_ENABLE = yes
+OLED_ENABLE = no
+RGB_MATRIX_ENABLE = yes
+RGB_MATRIX_DRIVER = WS2812
+LTO_ENABLE = yes
+
```
Comment (fauxpark) [inline keyboards/tominabox1/le_chiffre/rev2/rules.mk:18]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = yes           # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
ENCODER_ENABLE = yes
RGB_MATRIX_ENABLE = yes
RGB_MATRIX_DRIVER = WS2812
LTO_ENABLE = yes
```

**Review by fauxpark — CHANGES_REQUESTED** [review]
Please fix the file permissions and remove empty files.

**Comment by noroadsleft** [comment]
The file permissions and [fauxpark's note regarding the readme](https://github.com/qmk/qmk_firmware/pull/17377#discussion_r906855800) still need addressing.

**Comment by tominabox1** [comment]
> The file permissions and [fauxpark's note regarding the readme](https://github.com/qmk/qmk_firmware/pull/17377#discussion_r906855800) still need addressing.

I have already pushed an update to fauxpark's note and I'm not sure what file permission issues you're seeing.

File: keyboards/tominabox1/le_chiffre/readme.md [hunk 17]
```diff
@@ -4,12 +4,18 @@
...
-* Hardware Supported: Le Chiffre (source not available at this time)
+* Hardware Supported: [Le Chiffre](https://github.com/tominabox1/Le-Chiffre-Keyboard)
 * Hardware Availability: Bring your own case.
 
 Make example for this keyboard (after setting up your build environment):
 
-    make tominabox1/le_chiffre:default
+Rev 1:    make tominabox1/le_chiffre/rev1:default
+Rev 2:    make tominabox1/le_chiffre/rev2:default
+HE:       make tominabox1/le_chiffre/he:default
```
Comment (noroadsleft) [inline keyboards/tominabox1/le_chiffre/readme.md:19]:
```suggestion
    make tominabox1/le_chiffre/rev1:default
    make tominabox1/le_chiffre/rev2:default
    make tominabox1/le_chiffre/he:default
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
I'm still trying to find a solution for the file permissions. Git tracks them, but it's a feature Windows doesn't use at all, so figuring out how to set them is proving difficult for me.

----

**Comment by noroadsleft** [comment]
> I have already pushed an update to fauxpark's note and I'm not sure what file permission issues you're seeing.

GitHub is showing a lot of the files in this PR as follows:

image

The `100644 -> 100755` part at the end means the file permissions have changed. This affects how the files are handled in Linux (not sure about macOS; Windows doesn't use this feature at all).

**Comment by fauxpark** [comment]
`git update-index --chmod=-x filename`

**Comment by noroadsleft** [comment]
> `git update-index --chmod=-x filename`

Odd; I tried this command yesterday and it didn't do anything, but I try it now and it works. :confused: :man_shrugging:

**Comment by tominabox1** [comment]
I ran the command and updated but I dont know how to commit them. I did `git add .` then `git commit` and it said there was nothing to commit... even did the other suggested change and pushed it as you can see but it didnt fix the perms

**Comment by tominabox1** [comment]
Sorry think I got em now! I wish I understood how that happens

File: keyboards/tominabox1/le_chiffre/readme.md [hunk 18]
```diff
@@ -4,12 +4,18 @@
...
-* Hardware Supported: Le Chiffre (source not available at this time)
+* Hardware Supported: [Le Chiffre](https://github.com/tominabox1/Le-Chiffre-Keyboard)
 * Hardware Availability: Bring your own case.
 
 Make example for this keyboard (after setting up your build environment):
 
-    make tominabox1/le_chiffre:default
+make tominabox1/le_chiffre/rev1:default
+make tominabox1/le_chiffre/rev2:default
+make tominabox1/le_chiffre/he:default
```
Comment (fauxpark) [inline keyboards/tominabox1/le_chiffre/readme.md:19]:
```suggestion
    make tominabox1/le_chiffre/rev1:default
    make tominabox1/le_chiffre/rev2:default
    make tominabox1/le_chiffre/he:default 
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by noroadsleft — APPROVED** [review]
Looks good to me.

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**